proto/destination.proto

```diff
@@ -103,12 +103,16 @@ message WeightedAddr {
   AuthorityOverride authority_override = 7;
 }
 
-// Which strategy should be used for verifying TLS.
 message TlsIdentity {
   reserved 2;
   reserved "k8s_pod_identity";
 
-  oneof strategy { DnsLikeIdentity dns_like_identity = 1; }
+  oneof strategy {
+    DnsLikeIdentity dns_like_identity = 1;
+    UriLikeIdentity uri_like_identity = 3;
```

Comment:
If I recall correctly, the name _DnsLikeIdentity_ was chosen to indicate that this isn't actually a DNS name:

1. We don't expect it to be resolvable via DNS.
2. It doesn't actually support all DNS names (i.e., trailing dots are forbidden).

I think in this case, we can just call it UriIdentity.